connectors/src/api/auto_add_slack_channels.ts

```diff
@@ -0,0 +1,82 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Request } from "express";
+
+import { joinChannel } from "@connectors/connectors/slack/lib/channels";
+import { getSlackClient } from "@connectors/connectors/slack/lib/slack_client";
+import {
+  SlackChannel,
+  SlackConfigurationModel,
+} from "@connectors/lib/models/slack";
+import type { Logger } from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+export function isChannelNameWhitelisted(
+  remoteChannelName: string,
+  whiteListedChannelPatterns?: string
+): boolean {
+  if (!whiteListedChannelPatterns) {
+    return false;
+  }
+
+  const regex = new RegExp(whiteListedChannelPatterns);
+  return regex.test(remoteChannelName);
+}
+
+export async function autoJoinChannel(
+  req: Request,
```

Comment:
`nit` given that this function could be used from other places, I'd explicitly pass all the needed parameters and avoid exposing the request object.

Reply:
yes let's have the req unpacking in the main file and the implemetnation here

The file path is likely wrong as well should be under `src/connectors/slack/`